File: clojure/acronym/src/acronym.clj
(ns acronym
  (:require [clojure.string :as str]))

(defn acronym [s]
  (->> (-> s (str/split #"(\W+|[a-z](?=[A-Z]))"))
       (map first)
       (str/join)
       (str/upper-case)))

File: elixir/acronym/lib/acronym.ex
defmodule Acronym do
  def abbreviate(string) do
    string
    |> String.split(~r/[\s-_]+/, trim: true)
    |> Enum.map(&String.first/1)
    |> Enum.join("")
    |> String.upcase()
  end
end
